python/lsst/verify/compatibility/metricTask.py

```diff
@@ -0,0 +1,196 @@
+    datasets, regardless of the expected metric granularity, unless it is
+    impossible for more than one copy of a dataset to exist.
+
+    All input data shall be considered optional in order to maximize the
+    metric's usefulness for incomplete pipeline runs or runs with optional
+    processing steps. If a metric cannot be calculated because the necessary
+    inputs are missing, the ``MetricTask`` must return `None` in place of
+    the measurement.
+
+    .. note::
+        The API is designed to make it easy to convert all ``MetricTasks`` to
+        `~lsst.pipe.base.PipelineTask` later, but this class is *not* a
+        `~lsst.pipe.base.PipelineTask` and will not work with activators,
+        quanta, or `lsst.daf.butler`.
+
+    Parameters
+    ----------
+    Same as for `lsst.pipe.base.PipelineTask`.
+    """
+
+    # There may be a specialized MetricTaskConfig later, details TBD
+    ConfigClass = lsst.pex.config.Config
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataId):
+        """Compute a metric from in-memory data.
+
+        This implementation calls `run` on the contents of ``inputData``,
+        followed by calling `addStandardMetadata` on the result before
+        returning it. Any subclass that overrides this method must also call
+        `addStandardMetadata` on its measurement before returning it.
+
+        Parameters
+        ----------
+        inputData : `dict` from `str` to any
+            Dictionary whose keys are the names of input parameters and values
+            are Python-domain data objects (or lists of objects) retrieved
+            from data butler. Accepting lists of objects is strongly
+            recommended, to allow metrics to vary their granularity up to the
+            granularity of the input data without the need for extensive
+            code changes. Input objects may be `None` to represent
+            missing data.
+        inputDataIds : `dict` from `str` to `list` of dataId
+            Dictionary whose keys are the names of input parameters and values
+            are DataIds (or lists of DataIds) that task consumes for
+            corresponding dataset type. DataIds are guaranteed to match data
+            objects in ``inputData``
+        outputDataId : `dict` from `str` to dataId
+            Dictionary containing a single key-value pair, from
+            ``"measurement"`` to a single data ID for the measurement. The
+            data ID must have the same granularity as the metric.
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.Struct`
+            A ``Struct`` containing at least the following component:
+
+            - ``measurement``: the value of `getOutputMetric` derived from
+                               ``inputData`` (`lsst.verify.Measurement` or
+                               `None`). The measurement shall contain not only
+                               the value of the metric, but any additional
```

Comment:
"Metrics" don't have values (Measurements have values). You can say "The Measurement contains not only the value, but also metadata".

But this is true of `lsst.verify.Measurement` in general so I'm not sure it needs to be emphasized here.

Reply:
What I meant is that the method promises to handle any metadata, extras, etc. that may be mandatory, so that the caller doesn't have to. Suggested wording?

Reply:
Ok, I see. I think my suggested wording above works, otherwise keep as-is.